### PR TITLE
core: add ceph daemon command support

### LINKED ISSUE #208: is possible to support "ceph daemon cmd"?
such as: "ceph daemon osd.x dump_historic_ops"?

### PR DESCRIPTION
**Pull request by subhamkrai**
**Description of your changes:**
adding `ceph daemon *` command support

**Which issue is resolved by this Pull Request:**
Resolves #208

### CODE REVIEW

File: cmd/commands/ceph.go [hunk 1]
```diff
@@ -33,9 +36,20 @@ var CephCmd = &cobra.Command{
 	},
 	Run: func(cmd *cobra.Command, args []string) {
 		logging.Info("running 'ceph' command with args: %v", args)
-		_, err := exec.RunCommandInOperatorPod(cmd.Context(), clientSets, cmd.Use, args, operatorNamespace, cephClusterNamespace, false)
-		if err != nil {
-			logging.Fatal(err)
+		if args[0] == "daemon" {
+			if len(args) > 2 {
+				cephLabel := strings.Replace(args[1], "-", ".", 1)
```
Comment (travisn) [inline cmd/commands/ceph.go:41]: Could you factor out lines 41-42 to a helper method and then add unit tests for it? Anytime there is string parsing, I recommend unit tests.

Comment (travisn) [reply cmd/commands/ceph.go:41]: Also can you add a comment that shows the expected input? It's helpful to see why we need to replace a dash.

**Review by travisn — CHANGES_REQUESTED** [review]
How about an update to the readme with this new command?

**Comment by subhamkrai** [comment]
> How about an update to the readme with this new command?

there is already `ceph *` mentioned in Readme, do we want to add specific for `ceph daemon`?

**Comment by travisn** [comment]
> > How about an update to the readme with this new command?
> 
> there is already `ceph *` mentioned in Readme, do we want to add specific for `ceph daemon`?

Seems good to mention as a separate item because it fundamentally does something different. Instead of executing the ceph command in the operator pod, it would connect to the daemon pod.

File: README.md [hunk 2]
```diff
@@ -50,11 +50,12 @@ These are args currently supported:
     kubectl rook-ceph --context=$(kubectl config current-context) mons
     ```
 
-
 ### Commands
 
 - `ceph <args>` : Run a Ceph CLI command. Supports any arguments the `ceph` command supports. See [Ceph docs](https://docs.ceph.com/en/pacific/start/intro/) for more.
 
+- `ceph daemon <cephDaemon.id> <args>`: Run a Ceph daemon command connecting to the admin-socket.
```
Comment (travisn) [inline README.md:57]:
```suggestion
- `ceph daemon <daemon.id> <args>`: Run a Ceph daemon command by connecting to its admin socket.
```

File: cmd/commands/ceph.go [hunk 3]
```diff
@@ -33,9 +36,25 @@ var CephCmd = &cobra.Command{
 	},
 	Run: func(cmd *cobra.Command, args []string) {
 		logging.Info("running 'ceph' command with args: %v", args)
-		_, err := exec.RunCommandInOperatorPod(cmd.Context(), clientSets, cmd.Use, args, operatorNamespace, cephClusterNamespace, false)
-		if err != nil {
-			logging.Fatal(err)
+		if args[0] == "daemon" {
+			if len(args) > 2 {
```
Comment (travisn) [inline cmd/commands/ceph.go:40]: If there are not two args do they get an error?

File: cmd/commands/ceph.go [hunk 4]
```diff
@@ -33,9 +36,25 @@ var CephCmd = &cobra.Command{
...
+				cephDaemonNameAndID := splitCephDaemonNameAndID(args[1])
```
Comment (travisn) [inline cmd/commands/ceph.go:41]: Is the error handled if they don't include a `.`?

File: cmd/commands/ceph.go [hunk 5]
```diff
@@ -33,9 +36,25 @@ var CephCmd = &cobra.Command{
...
+			if err != nil {
+				logging.Fatal(err)
+			}
 		}
 	},
 }
+
+// splitCephDaemonNameAndID splits the arg based on '.' so that we use it with label, example osd.0 to osd and 0
+func splitCephDaemonNameAndID(args string) []string {
+	cephDaemonNameAndID := strings.Split(args, ".")
```
Comment (travisn) [inline cmd/commands/ceph.go:58]:
```suggestion
	return strings.Split(args, ".")
```

File: docs/ceph.md [hunk 6]
```diff
@@ -96,3 +96,18 @@ kubectl rook-ceph ceph status --format json-pretty
 #     "progress_events": {}
 # }
 ```
+
+## Daemon
+
+Run Ceph daemon command connecting to the admin-socket.
```
Comment (travisn) [inline docs/ceph.md:102]:
```suggestion
Run the Ceph daemon command by connecting to its admin socket.
```

**Review by travisn — CHANGES_REQUESTED** [review]
A few small suggestions

File: cmd/commands/ceph.go (hunk 3 above)
Comment (subhamkrai) [reply cmd/commands/ceph.go]: now it has
```bash
./bin/kubectl-rook-ceph ceph daemon
Warning: rook version 'rook: v1.14.0-alpha.0.80.gf84e932ec' is running a pre-release version of Rook.

Info: running 'ceph' command with args: [daemon]
Error: too few arguments to run the 'ceph daemon' command
```

File: cmd/commands/ceph.go (hunk 4 above)
Comment (subhamkrai) [reply cmd/commands/ceph.go]: now it has
```bash
./bin/kubectl-rook-ceph ceph daemon osd 0
Warning: rook version 'rook: v1.14.0-alpha.0.80.gf84e932ec' is running a pre-release version of Rook.

Info: running 'ceph' command with args: [daemon osd 0]
Error: Invalid argument to 'ceph daemon' command: osd. The arg should be in the format of '<daemon.id>'
```

**Review by travisn — APPROVED** [review]

**Pull request merged**